include/seqan3/search/fm_index/bi_fm_index.hpp

```diff
@@ -94,13 +95,22 @@ class bi_fm_index
 
 public:
 
+    static_assert(dimension_v<text_t> == 1 || dimension_v<text_t> == 2,
+                  "Only texts or collections of texts can be indexed.");
+
+    //!\brief Indicates whether index is built over a collection.
+    static bool constexpr is_collection = dimension_v<text_t> == 2;
+
     /*!\name Text types
      * \{
      */
     //!\brief The type of the forward indexed text.
     using text_type = text_t;
     //!\brief The type of the forward indexed text.
-    using rev_text_type = decltype(view::reverse(*text));
+    using rev_text_type = std::conditional_t<is_collection,
+                                             decltype(*text | view::deep{view::reverse} | view::deep{view::persist}
+                                                            | view::reverse),
```

Comment:
Because for the bi_fm_index we need the text and the reversed text.
Example (ignoring the rank stuff)

text = `[ACGT, TGC]`
delimiter = `#`
flattened text = `ACGT#TGC`
reversed text = `CGT#TGCA`

text | deep_reverse = `[TGCA,CGT]`
^ | reverse = `[CGT,TGCA]`
^ flattened = `CGT#TGCA`

Reply:
I know that the bifm needs the text in both direction, but why does "in both directions" mean that you need to reverse the order? I don't think this is being done in SeqAn2, is it?

Reply:
The sdsl bi_fm works on single texts, where one is forward and one is reversed. If we do not reverse the second one after deep_reversing it, it's not the true reverse of the text (seen as single sequence) and hence the synchronisation between both directions will fail.

Reply:
Hm, yeah, I understand. :+1: